wezterm-font/src/locator/mod.rs

```diff
@@ -44,6 +44,12 @@ pub enum FontDataSource {
     },
 }
 
+impl std::hash::Hash for FontDataSource {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        core::mem::discriminant(self).hash(state);
```

Comment:
This feels like a net negative change: why remove the derived Hash in favor of whatever this is doing?

Reply:
The the partial eq and hash has to match. Making both explicit should atleast make sure that we ignore fields equally, etc.
This should be equivalent to the derived result.